pkg/block/s3/adapter.go

```diff
@@ -308,7 +308,7 @@ func (a *Adapter) Get(ctx context.Context, obj block.ObjectPointer, _ int64) (io
 		log.WithError(err).Errorf("failed to get S3 object bucket %s key %s", qualifiedKey.StorageNamespace, qualifiedKey.Key)
 		return nil, err
 	}
-	sizeBytes = *objectOutput.ContentLength
+	sizeBytes = aws.Int64Value(objectOutput.ContentLength)
```

Comment:
This looks good, but TBH I don't understand what this line does: it copies the value into a variable, which it then ignores.  `sizeBytes` is not a pointer, I don't see what effect this assignment can have.
```suggestion
```

(I'm actually shocked that Go doesn't flag an unused variable here.  It's the sort of thing gcc has been doing for ages now, at least when optimizing.)

Same applies below, too.   It looks like a bad leftover.

Reply:
The variable is referenced in the defer function - so it is just for report the file size as part of the metric.